pkgs/applications/graphics/qosmic/default.nix

```diff
@@ -0,0 +1,64 @@
+{ mkDerivation
+, fetchFromGitHub
+, fetchpatch
+, qmake
+, wrapQtAppsHook
+, qtbase
+, pkg-config
+, lua
+, flam3
+, libxml2
+, libpng
+, libjpeg
+, lib
+, }:
+
+mkDerivation rec {
+  pname = "qosmic";
+  version = "1.6.0";
+
+  src = #/home/aengelen/dev/qosmic;
+  fetchFromGitHub {
+    owner = "bitsed";
+    repo = "qosmic";
+    rev = "v${version}";
+    sha256 = "13nw1mkdib14430r21mj352v62vi546vf184vyhxm7yjjygyra1w";
+  };
+
+  patches = [
+    ./0000-use-pkg-config.patch
+    (fetchpatch {
+      name = "fix-class-QButtonGroup-include-errors-with-Qt-5.11.patch";
+      url = "https://github.com/bitsed/qosmic/commit/3f6e1ea8d384a124dbc2d568171a4da798480752.patch";
+      sha256 = "0bp6b759plkqs32nvfpkfvf3qqzc9716k3ycwnjvwabbvpg1xwbl";
+    })
+  ];
+
+  nativeBuildInputs = [ qmake wrapQtAppsHook pkg-config ];
+  buildInputs = [
+    qtbase
+    lua
+    flam3
+    libxml2
+    libpng
+    libjpeg
+  ];
+  installPhase = ''
+    runHook preInstall
+
+    INSTALL_ROOT=$out make install
+    mv $out/usr/bin $out/bin
+    mv $out/usr/share $out/share
```

Comment:
Maybe change

https://github.com/bitsed/qosmic/blob/43c96ab7bf6a7a5c59767c04d6d7868350cac239/qosmic.pro#L5

to be overridable as we did in

https://github.com/corrados/jamulus/pull/247/files#diff-ff31b6d635c626acb697d69ef0e864bdR305-R307

Then our qmakeFlags will work and the whole `installPhase` will be unnecessary.

https://github.com/NixOS/nixpkgs/blob/fc00e14704dc29d16b2e2c0603531ffd04f527a2/pkgs/development/libraries/qt-5/hooks/qmake-hook.sh#L6-L7

Reply:
That's indeed a great cleanup! Proposed it upstream in https://github.com/bitsed/qosmic/pull/39 and in the mean time added that commit as a patch here.